zipkin-storage/elasticsearch-http/src/main/java/zipkin/storage/elasticsearch/http/CallbackListenableFuture.java

```diff
@@ -40,13 +40,28 @@
     call.cancel();
   }
 
-  @Override public void onResponse(Call call, Response response) throws IOException {
+  /** Note: this runs on the {@link okhttp3.OkHttpClient#dispatcher() dispatcher} thread! */
+  @Override public void onResponse(Call call, Response response) {
     try (ResponseBody responseBody = response.body()) {
       if (response.isSuccessful()) {
         set(convert(responseBody));
       } else {
         setException(new IllegalStateException("response failed: " + response));
       }
+    } catch (Throwable t) {
```

Comment:
running against mixed versions increases the sorts of crazy exceptions. so tightened this and backfilled exception test cases.